### DISCUSSION TITLE
Question: spawn/child_process options

### DISCUSSION THREAD

**Discussion by rmcarias** [google/zx#371] [Help needed]
## Specifications
This is more of question (or feature request). How can I pass the supported options to the spawn method for example: 
```
{
  detached: true,
  stdio: 'inherit',
  shell: true,
}
```
Using this syntax?
```$`<some_shell_command>````

Also, it seems that when you do this:
```
let child = $`<some_shell_command>`
```
Because it returns a promise, we don't get access to other child process props and method, like
```
child.unref()
```

- Version: 6.0.7
- Platform: Mac OS

**Comment by antonmedv** [comment]
> stdio: 'inherit',

This already is setup. But will change depending on usage of .pipe method.

>   shell: true,

Same. This is on already.

>   detached: true,

This can't be changed.

unref is accessable:
```js
let p = $`<some_shell_command>`
p.child.unref()
```

If some options of spawn not available yet - better to use low level api directly.

**Reply by rmcarias** [reply]
@antonmedv  Thank you this answers my question. I believe I can just also use `$.spawn()` if I want to use the raw API correct?

**Reply by antongolub** [reply]
You may wrap `spawn` fn to capture and override its arguments:
```js
spawn = $.spawn // defaults to child_process.spawn
$.spawn = (cmd, opts) => spawn(cmd, {
   ...opts,
   stdio: ['pipe', 'pipe', 'pipe'],
})
```

**Reply by rmcarias** [reply]
Thanks. That works @antongolub
